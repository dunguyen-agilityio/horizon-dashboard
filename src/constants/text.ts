export const FOOTER_CONTENT =
  'Â© 2022 Horizon UI. All Rights Reserved. Made with love by Simple!';
