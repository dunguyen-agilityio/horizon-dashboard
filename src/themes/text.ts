export const fontSizes = {
  xs: '0.625rem', //10px
  tiny: '0.75rem', // 12px
  small: '0.875rem', // 14px
  medium: '1rem', // 16px
  large: '1.125rem', // 18px
  xl: '1.625rem', //  26px
  extra: '1.375rem', //22px
  '2xl': '2.125rem', // 34px
  '3xl': '2.5rem', //  40px
};

export const lineHeight = {
  tiny: '1rem', // 16px
  small: '1.25rem', //  20px
  medium: '1.5rem', //  24px
  large: '1.75rem', //  28px
  extra: '1.625px', //  26px
  '2xl': '2.625rem', // 42px
  xl: '2rem', // 32px
};
