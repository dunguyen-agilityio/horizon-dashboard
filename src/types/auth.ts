import { User } from '@/models/User';

export interface SignInFormData {
  identifier: string;
  password: string;
}

export interface SignUpFormData extends Pick<SignInFormData, 'password'> {
  userName: string;
  email: string;
  confirmPassword: string;
}

export type SignInResponse = {
  jwt: string;
  user: User;
};

export type ResetPasswordFormData = {
  email: string;
};
