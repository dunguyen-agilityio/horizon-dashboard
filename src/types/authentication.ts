export type SignInForm = {
  identifier: string;
  password: string;
};
