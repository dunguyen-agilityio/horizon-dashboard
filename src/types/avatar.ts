export interface IAvatar {
  id: number;
  name: string;
  url: string;
  createdAt: string;
  updatedAt: string;
}

export interface IAvatarData {
  data: {
    attributes: IAvatar;
  };
}
