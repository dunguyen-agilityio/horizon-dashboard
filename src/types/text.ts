export const enum TEXT_SIZE {
  '2xl' = '2xl',
  xl = 'xl',
  extra = 'extra',
  lg = 'lg',
  md = 'md',
  sm = 'sm',
  xs = 'xs',
}

export const enum TEXT_VARIANT {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
}
