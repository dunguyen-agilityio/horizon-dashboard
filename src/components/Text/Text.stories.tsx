import type { Meta, StoryObj } from '@storybook/react';

import Text from '.';
import { TEXT_SIZE, TEXT_VARIANT } from '@/types/text';

const meta = {
  title: 'Components/Text',
  component: Text,
  tags: ['autodocs'],
} satisfies Meta<typeof Text>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Medium: Story = {
  args: {
    text: 'Medium',
  },
};

export const ExtraLarge2: Story = {
  args: {
    size: TEXT_SIZE['2xl'],
    text: '2 Extra Large',
  },
};

export const ExtraLarge: Story = {
  args: {
    size: TEXT_SIZE.xl,
    text: 'Extra Large',
  },
};

export const Large: Story = {
  args: {
    size: TEXT_SIZE.lg,
    text: 'Large',
  },
};

export const Small: Story = {
  args: {
    size: TEXT_SIZE.sm,
    text: 'Small',
  },
};

export const ExtraSmall: Story = {
  args: {
    size: TEXT_SIZE.xs,
    text: 'Extra Small',
  },
};

export const Primary: Story = {
  args: {
    text: 'Primary',
  },
};

export const Secondary: Story = {
  args: {
    text: 'Secondary',
    variant: TEXT_VARIANT.SECONDARY,
  },
};
