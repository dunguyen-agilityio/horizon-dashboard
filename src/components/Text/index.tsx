import { TEXT_SIZE, TEXT_VARIANT } from '@/types/text';
import { cn } from '@nextui-org/theme';

const sizes: { [key in TEXT_SIZE]: string } = {
  [TEXT_SIZE.xl]: 'text-xl font-bold leading-extra',
  [TEXT_SIZE['2xl']]: 'text-2xl font-bold leading-2xl',
  [TEXT_SIZE.lg]: 'text-extra font-bold leading-large',
  [TEXT_SIZE.md]: 'text-medium leading-xl',
  [TEXT_SIZE.sm]: 'text-small font-medium',
  [TEXT_SIZE.xs]: 'text-xs uppercase',
};

const variants: { [key in TEXT_VARIANT]: string } = {
  [TEXT_VARIANT.PRIMARY]: 'text-primary dark:text-white',
  [TEXT_VARIANT.SECONDARY]: 'text-secondary dark:text-secondary',
};

export interface TextProps {
  size?: TEXT_SIZE;
  variant?: TEXT_VARIANT;
  as?: keyof React.ReactHTML;
  className?: string;
}

const Text = ({
  children,
  size = TEXT_SIZE.md,
  as: Component = 'p',
  variant = TEXT_VARIANT.PRIMARY,
  className,
}: React.PropsWithChildren<TextProps>) => {
  return (
    <Component className={`${variants[variant]} ${cn(sizes[size], className)}`}>
      {children}
    </Component>
  );
};

export default Text;
