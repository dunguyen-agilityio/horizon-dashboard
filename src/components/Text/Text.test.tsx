import { render } from '@testing-library/react';
import Text, { type TextProps } from '.';
import { TEXT_SIZE, TEXT_VARIANT } from '@/types/text';

describe('Text tests', () => {
  const setup = (props: TextProps = {}) => render(<Text {...props} />);

  it('Should match snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot', () => {
    const { container } = setup({
      as: 'h1',
      size: TEXT_SIZE['2xl'],
      variant: TEXT_VARIANT.SECONDARY,
    });
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot', () => {
    const { container } = setup({
      as: 'h1',
      size: TEXT_SIZE['2xl'],
      variant: TEXT_VARIANT.SECONDARY,
    });
    expect(container).toMatchSnapshot();
  });
});
