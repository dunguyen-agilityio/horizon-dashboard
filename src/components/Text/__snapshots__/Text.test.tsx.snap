// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text tests Should match snapshot 1`] = `
<div>
  <p
    class="text-primary dark:text-white text-medium leading-xl"
  >
    Hello Wold!
  </p>
</div>
`;

exports[`Text tests Should match snapshot 2`] = `
<div>
  <h1
    class="text-secondary dark:text-secondary text-2xl font-bold leading-2xl"
  >
    Hello Wold!
  </h1>
</div>
`;

exports[`Text tests Should match snapshot 3`] = `
<div>
  <h1
    class="text-secondary dark:text-secondary text-2xl font-bold leading-2xl"
  >
    Hello Wold!
  </h1>
</div>
`;
